beater/server.go

```diff
@@ -68,34 +54,35 @@ func (s *Server) create(successCallback func([]beat.Event), host string) *http.S
 	}
 
 	return &http.Server{
-		Addr:           host,
+		Addr:           config.Host,
 		Handler:        mux,
-		ReadTimeout:    s.config.ReadTimeout,
-		WriteTimeout:   s.config.WriteTimeout,
-		MaxHeaderBytes: s.config.MaxHeaderBytes,
+		ReadTimeout:    config.ReadTimeout,
+		WriteTimeout:   config.WriteTimeout,
+		MaxHeaderBytes: config.MaxHeaderBytes,
 	}
 }
 
-func (s *Server) Start(successCallback func([]beat.Event), host string) {
-	s.http = s.create(successCallback, host)
-	if s.config.SSL.IsEnabled() {
-		go s.http.ListenAndServeTLS(s.config.SSL.Cert, s.config.SSL.PrivateKey)
+func start(server *http.Server, ssl *SSLConfig) {
+	if ssl.isEnabled() {
+		go server.ListenAndServeTLS(ssl.Cert, ssl.PrivateKey)
 	} else {
-		go s.http.ListenAndServe()
+		go server.ListenAndServe()
 	}
 }
 
-func (s *Server) Stop() error {
+func stop(server *http.Server) error {
 	c := context.TODO()
-	err := s.http.Shutdown(c)
+	err := server.Shutdown(c)
 	return err
 }
 
-func (s *Server) createHandler(p processor.Processor, successCallback func([]beat.Event)) func(w http.ResponseWriter, r *http.Request) {
+type handler func(w http.ResponseWriter, r *http.Request)
+
+func createHandler(p processor.Processor, config Config, publish successCallback) handler {
```

Comment:
It just abstracts which methods belong to one struct. I am not a fan of having one struct with a huge amount of methods, since this makes it really hard to test, but in this case it would be two methods that obviously need this struct to make any sense. 
TBH I am generally a bit conflicted on this topic.. maybe I can find some good reading/examples and then come back to this discussion a bit later.